e3/src/common/macros.rs

```diff
@@ -0,0 +1,23 @@
+macro_rules! unlock {
+    ($mutex:expr) => {{
+        $mutex.lock().expect("error taking the lock")
```

Comment:
There are a lot of locks across whole crate. I know I'm not expert on lock-free structure, but I'm afraid these frequent context switches may create more bottlenecks. We might want to consider more and maybe have some discussions on RFC.

Reply:
Definitely. Taking locks all the time in the **hot-path** is something that for sure can become a bottleneck. But I really doubt that taking locks is significant compared to many other time-consuming tasks the node has to do (hashing, signature verification, PoW, DB I/O, etc).